.github/workflows/gcp-full-infra-test-plan.yaml

```diff
@@ -0,0 +1,106 @@
+  pull_request:
+    branches: [main]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: "read"
+  id-token: "write"
+
+env:
+  PR_NUMBER: ${{ github.event.pull_request.number }}
+  tenant_group_name: "workflow-pr-${{ github.event.pull_request.number }}"
+  tenant_name: "tenant-1"
+
+jobs:
+  plan:
+    runs-on: ubuntu-latest
+    environment: testing
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set PR_NUMBER env variable
+        if: github.event_name == 'workflow_dispatch'
+        run: |
+          # If event is workflow_dispatch, use the event.workflow_dispatch.inputs.pr_number
+          echo "PR_NUMBER=${{ github.event.inputs.pr_number }}" >> $GITHUB_ENV
+          echo "tenant_group_name=workflow-pr-${{ github.event.inputs.pr_number }}" >> $GITHUB_ENV
+
+      - name: Setup GCloud
+        uses: "google-github-actions/auth@v2.1.1"
+        with:
+          workload_identity_provider: ${{ secrets.GCP_WORKLOAD_IDENTITY_PROVIDER }}
+          service_account: ${{ secrets.GCP_SERVICE_ACCOUNT }}
+
+      - name: Set up Cloud SDK
+        uses: "google-github-actions/setup-gcloud@v2.1.0"
+        with:
+          version: ">= 363.0.0"
+
+      - name: Set up Tofu
+        uses: opentofu/setup-opentofu@v1.0.1
+
+      - name: Plan tenant group
+        env:
+          TF_VAR_cluster_name: ${{ env.tenant_group_name }}
+          TF_VAR_tenant_group_name: ${{ env.tenant_group_name }}
+          TF_VAR_project_id: ${{ vars.GCP_PROJECT_ID }}
+          TF_VAR_force_destroy_backup_bucket: ${{ vars.GCP_FORCE_DESTROY_BACKUP_BUCKET }}
+          TF_VAR_region: ${{ vars.GCP_REGION }}
+          TF_VAR_subnet_cidr: ${{ vars.GCP_SUBNET_CIDR }}
+          TF_VAR_ip_range_pods: ${{ vars.GCP_IP_RANGE_PODS }}
+          TF_VAR_ip_range_services: ${{ vars.GCP_IP_RANGE_SERVICES }}
+          TF_VAR_tenant_provision_sa: ${{ vars.GCP_TENANT_PROVISION_SA }}
+          TF_VAR_backup_retention_policy_days: ${{ vars.GCP_BACKUP_RETENTION_POLICY_DAYS }}
+          TF_VAR_cluster_deletion_protection: ${{ vars.GCP_CLUSTER_DELETION_PROTECTION }}
+        working-directory: ./tofu/gcp/tenant_group
+        run: |
+          mkdir -p ${GITHUB_WORKSPACE}/artifacts
+          tofu init -backend-config="bucket=${{ vars.GCP_STATE_BUCKET_NAME }}" -backend-config="prefix=tenant_group/${{ env.tenant_group_name }}"
+          tofu test
+          tofu plan -out ${GITHUB_WORKSPACE}/artifacts/tenant_group.tfplan
+
+      - name: Plan tenant
+        env:
+          TF_VAR_project_id: ${{ vars.GCP_PROJECT_ID }}
+          TF_VAR_region: ${{ vars.GCP_REGION }}
+          TF_VAR_vpc_name: "vpc"
+          TF_VAR_cluster_name: "some-cluster"
+          TF_VAR_cluster_ca_certificate: "dGVzdA=="
+          TF_VAR_cluster_endpoint: "255.255.255.255"
+          TF_VAR_backup_bucket_name: "backup-bucket"
+          TF_VAR_ip_address: "255.255.255.255"
```

Comment:
are you sure you don't want to move this to env vars?

Reply:
These values are going to be replaced in the `apply` step with the values in the output of the `tenant_group`. They are here only because we can't plan the `tenant` without them.